### DISCUSSION TITLE
How to generate a deps.json file

### DISCUSSION THREAD

**Discussion by KOosterhuis** [dotnet/roslyn#61080] [Q&A]
We are dynamically generating source code C#.
And we use CSharpCompilation (From the nuget: Microsoft.CodeAnalysis)  to emit the binary/pdf streams based on parsed syntax trees and assembly references.

We store this on disk and then we can run the program's.

However since the move to .net 6 programs break because they can't find System.Management.dll at runtime. This is no longer stored in the program folder but in a sub folder  ".\runtimes\win\lib\net6.0\System.Management.dll".

When building from visual studio it generates a program.deps.json file which redirects to this sub folder. 

How can we use the Microsoft Roslyn to also generate this deps.json file?
Just pasting the parts needed to System.Management.dll doesn't work as then the normal assemblies in the program folder can't be found anymore.

**Comment by MSDN-WhiteKnight** [comment]
I think Roslyn is not aware of SDK .deps.json format, but there's Microsoft.Extensions.DependencyModel library ([DependencyContextWriter](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencymodel.dependencycontextwriter?view=dotnet-plat-ext-6.0)) that can be used to process it. It's mostly not documented, but you can find examples in source code where it's used like https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs#L126 

It might be simpler to hand-generate JSON by example of what SDK generates, though.
